src/client/Microsoft.Identity.Client/Internal/Logger/LoggerAdapterHelper.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.PlatformsCommon.Factories;
+using Microsoft.IdentityModel.Abstractions;
+
+namespace Microsoft.Identity.Client.Internal.Logger
+{
+    internal class LoggerAdapterHelper
+    {
+        private static Lazy<string> s_msalVersionLazy = new Lazy<string>(() => MsalIdHelper.GetMsalVersion());
+        private static Lazy<string> s_runtimeVersionLazy = new Lazy<string>(() => PlatformProxyFactory.CreatePlatformProxy(null).GetRuntimeVersion());
+        private static readonly Lazy<ILoggerAdapter> s_nullLogger = new Lazy<ILoggerAdapter>(() => new NullLogger());
+
+        public static string GetClientInfo(string clientName, string clientVersion)
+        {
+            string clientInformation = string.Empty;
+            if (!string.IsNullOrEmpty(clientName) && !ApplicationConfiguration.DefaultClientName.Equals(clientName))
+            {
+                // space is intentional for formatting of the message
+                if (string.IsNullOrEmpty(clientVersion))
+                {
+                    clientInformation = string.Format(CultureInfo.InvariantCulture, " ({0})", clientName);
+                }
+                else
+                {
+                    clientInformation = string.Format(CultureInfo.InvariantCulture, " ({0}: {1})", clientName, clientVersion);
+                }
+            }
+
+            return clientInformation;
+        }
+
+        public static ILoggerAdapter CreateLogger(
+            Guid correlationId,
+            ApplicationConfiguration config)
+        {
+            if (config.IdentityLogger == null)
+            {
+                return LegacyIdentityLoggerAdapter.Create(correlationId, config);
```

Comment:
Should this be renamed to DefaultLogger instead?

Reply:
technically the nullLogger is the default logger. the LegacyIdentityLoggerAdapter is only used when the developer sets it.